### DISCUSSION TITLE
Set vtype and calculate sDA

### DISCUSSION THREAD

**Discussion by Soushiyant17** [LBNL-ETA/pyradiance#39] [Q&A]
@taoning 
Hi dear Taoning,

I've been working with your interesting library for the past few weeks, but I've encountered several issues and would appreciate your guidance.

1.When I tried to change the view type in the `pr.View()` function using `pr.model.ViewType`, there was no change in the output. Only `pr.model.ViewType.VT_CYL` seemed to affect the results. How can I set the view type to Fisheye?

2.Is there a way to simulate annual metrics using Pyradiance? For example, if we want to calculate the sDA, how should we approach this? rfluxmtx? I also checked the Frads library, where this was done by calling the IDF model. However, I want to accomplish it using Pyradiance. Would you be able to share a sample code? Assuming the user is present in the space from 8 AM to 6 PM.

3.I am very eager for the new Radiance update so that I can also perform spectral simulations. How long do you think it will take to add these capabilities to the libraries?
And will your presentation at this year's workshop be recorded and placed on the website, or will only your presentation file be made available?

best regards,
Soushiyant

**Comment by taoning** [comment]
Hi Soushiyant, thanks for trying it out.
1. Sorry the `pr.model.ViewType` is not in use, please use string to define view type, e.g., 'a', 'h'.   This will be fixed later.
2. Yes, running annual simulation using pyradiance is doable, the workflow would be similar to running Radiance CLI.  You'd need to manually construct the `sender` and `receiver` and call `rfluxmtx`. Please also note that `frads` doesn't require you to use EnergyPlus model just to simulate annual daylight.  (`frads` was developed to help with annual daylight simulation in the first place and pyradiance was meant to just reflect Radiance CLI.) [Here](https://lbnl-eta.github.io/frads/how-to/guide_rad3/) is a walk-through using three-phase method, we should include more tutorials soon.
3. We should have the all of the spectral feature in place this week.

I don't know if the workshop would be recorded.

**Reply by Soushiyant17** [reply]
Thank you for your response.

1. Thank you, it worked with string,  but when I changed -vh and -vv return bad header, Is it possible to check it?

2.This method seems somewhat complicated. I tried to do it based on what was presented [here](https://lbnl-eta.github.io/frads/ref/matrix/), but I couldn't succeed :( need more tutorials

3.Great! I am looking forward to the new version.

4.The process presented [here](https://lbnl-eta.github.io/frads/how-to/guide_rad3/) works well for me, but when I try to apply it to method 1, I encounter an error, while the same input files work correctly in method 2.
![pr_m1](https://github.com/user-attachments/assets/82d35c1b-1084-4a7f-89ea-57d41a2d9203)

**Reply by taoning** [reply]
Can you provide your code and error in text so that I can run it easily?   More detail is better.  Some of these might be bug, but I 'd like to see more code. (If you suspect it might  be a bug, please raise an issue instead). Thanks.

I am gonna be out until the week after next, so the reply might be delayed.
